extensions/amp-form/0.1/form-submit-service.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+export class FormSubmitService {
+  /**
+   * Global service used to register callbacks we wish to execute when an
+   * amp-form is submitted.
+   */
+  constructor() {
+    /** @private {!Array} */
+    this.callbacks_ = [];
+  }
+
+  /**
+   * Used to register callbacks.
+   * @param {function(*)} cb
+   */
+  beforeSubmit(cb) {
```

Comment:
this is good for now. we can separately introduce an async version later when needed.

Reply:
SGTM